pymc3/variational/approximations.py

```diff
@@ -0,0 +1,526 @@
+import collections
+import theano.tensor as tt
+from theano.ifelse import ifelse
+import theano
+import pymc3 as pm
+from ..theanof import tt_rng, memoize, change_flags
+from ..blocking import ArrayOrdering
+from ..distributions.dist_math import rho2sd, log_normal3
+from ..math import flatten_list
+from ..model import modelcontext
+import numpy as np
+
+# helper class
+FlatView = collections.namedtuple('FlatView', 'input, replacements')
+# shortcut for zero grad
+Z = theano.gradient.zero_grad
+
+
+class BaseApproximation(object):
+    """Base class for variational methods
+    It uses the idea that we can map standard normal distribution
+    to the approximate posterior distribution. Thus there are three blank
+    places in the implementation:
+        1) creating shared parameters `create_shared_params()`
+        2) mapping initial distribution to posterior `posterior{_{global|local}}(initial)`
+        3) calculating log_q_W for approximation `log_q_W{_{global|local}}(posterior)`
+
+    Then ELBO can be calculated for given approximation
+        log_p_D(posterior) - KL_q_p_W(posterior)
+    """
+    def __init__(self, population=None, known=None, model=None):
```

Comment:
Class docstring is always used to document init method(jupyter even has a popup help with shift+tab). All parent classes are supposed to be documented and thus will have docstrings for init. So I'll update a class docstring with signature description